text/0314-cdk-construct-hub.md

```diff
@@ -0,0 +1,278 @@
+
+## FAQ
+
+### What are we launching today?
+
+CDK Construct Hub is a new website that makes it easy for developers to
+discover, use, and share CDK constructs that encapsulate cloud infrastructure
+patterns. The community has already published hundreds of CDK constructs on the
+public registries like npmjs.com, PyPI, Maven Central, and NuGet. However those
+are distributed repositories, managed by numerous organizations, and lack
+centralized search tools sufficient for developers to easily find the cloud
+infrastructure patterns they are looking for, in the programming language they
+want to use.
+
+CDK Construct Hub automatically indexes all CDK construct libraries published to
+the various general purpose package managers, so that developers have a single
+place to search for constructs, regardless of preferred programming language or
+provisioning engine. Developers will find constructs published by AWS, partners,
+and individual developers. CDK Constuct Hub also generates unified documentation
+for the constructs in all supported programming languages.
+
+CDK Construct Hub is also available as a deploymable CDK construct library, so
+that enterprises and users with specific needs can host a private instance of
+the catalog, making interally developed constructs easier for teams to find.
+Just like the public instance of CDK Construct Hub scans public repositories for
+construct libraries, the private CDK Construct Hub instance can be configured to
+scan private repositories like AWS CodeArtifact, JFrog Artifactory, and others.
+
+### Why should I use this feature?
+
+The CDK Construct Hub is a centralized place that helps discover and share
+custom-made constructs compatible with the AWS Cloud Development Kit (CDK), CDK
+for Kubernetes (CDK8s), or CDK for Terraform (CDK-Tf). It shortens the time it
+takes to ship cloud applications by allowing developers to discover constructs
+that address their infrastructure requirements, and provides a unified
+cross-language documentation experience to help developers integrate those
+constructs into their application.
+
+Customers can also deploy a private instance of CDK Construct Hub to facilitate
+sharing infrastructure abstractions published to a private package registry
+within the organization.
+
+### Where can I find the CDK Construct Hub?
+
+The public instance of CDK Construct Hub is at: [awscdk.io](https://awscdk.io/).
+
+### How do I publish a package to CDK Construct Hub?
+
+Packages published to the public npm registry (npmjs.com) are automatically
+indexed by the public instance of CDK Construct Hub as long as they meet the
+following prerequisites:
+
+1. Cross-language support has been configured correctly using [jsii]
+1. The `cdk` keyword is referenced (in the `keywords` key of `package.json`)
+
+[jsii]: https://aws.github.io/jsii
+
+### Can I opt packages out of the public instance of CDK Construct Hub?
+
+There is currently no self-service way to un-list a package from the CDK
```

Comment:
Why? We should be able to easily support this through an additional label

Reply:
But you cannot add a label to an _already_ published version (those are immutable)... The corollary is if _any_ version of a package has the label, then _all_ versions of the package are opted out, forever... This does not allow library authors to change their minds in the future...

Reply:
We can decide that the labels on the latest version will determine if the module appears or not but we can leave this detail for later.

Reply:
define configured correctly? like its in package.json or it actually working? Or its on ALL languages or 1